SimG4Core/Notification/interface/G4SimTrack.h

```diff
@@ -80,6 +107,15 @@ class G4SimTrack {
   math::XYZVectorD parentMomentum_;
   math::XYZVectorD tkSurfacePosition_;
   math::XYZTLorentzVectorD tkSurfaceMomentum_;
+  bool crossedBoundary_;
+  int idAtBoundary_;
+  math::XYZVectorD positionAtBoundary_;
+  math::XYZTLorentzVectorD momentumAtBoundary_;
+  void assertCrossedBoundary() const {
```

Comment:
I would remove this assert. The track may cross or not cross a boundary - nothing more.

Reply:
I have removed the assert method.